include/seqan/journaled_string_tree/journaled_string_tree_traverser.h

```diff
@@ -266,165 +261,251 @@ struct Member<TraverserImpl<TJst, JstTraversalSpec<TSpec>, TObserver>, Traverser
+
+template <typename TJst, typename TSpec, typename TSize>
+inline void
+setContextSize(TraverserImpl<TJst, JstTraversalSpec<TSpec> > & me,
+               TSize contextSize)
+{
+    me._needInitialization = true;
+    me._contextSize = contextSize;
+}
+
+// ----------------------------------------------------------------------------
+// Function branchSize();
+// ----------------------------------------------------------------------------
+
+template <typename TJst, typename TSpec>
+inline auto
+branchSize(TraverserImpl<TJst, JstTraversalSpec<TSpec> > const & me) -> decltype(me._branchLength)
+{
+    return me._branchLength;
+}
+
+// ----------------------------------------------------------------------------
+// Function setBranchSize();
+// ----------------------------------------------------------------------------
+
+template <typename TJst, typename TSpec, typename TSize>
+inline void
+setBranchSize(TraverserImpl<TJst, JstTraversalSpec<TSpec> > & me,
+              TSize newSize)
+{
+    me._needInitialization = true;
+    me._branchLength = newSize;
+}
+
+// ----------------------------------------------------------------------------
+// Function position();
+// ----------------------------------------------------------------------------
+
+template <typename TJst, typename TSpec>
+inline typename Position<TraverserImpl<TJst, JstTraversalSpec<TSpec> > >::Type
+position(TraverserImpl<TJst, JstTraversalSpec<TSpec> > const & me)
+{
+
+    if (length(impl::stack(me)) == 1)
+        return impl::positionReference(me);
+    else
+        return impl::positionBranch(me);
+}
+
+// ----------------------------------------------------------------------------
+// Function init();
+// ----------------------------------------------------------------------------
+
+template <typename TJst, typename TSpec,
+          typename TObserver,
+          typename TProxySelector>
+inline void
+init(TraverserImpl<TJst, JstTraversalSpec<TSpec> > & me,
+     TObserver & observer,
+     Tag<TProxySelector> const & /*tag*/)
+{
+    impl::init(me, observer, Tag<TProxySelector>());
+    me._needInitialization = false;
+}
+
+template <typename TJst, typename TSpec,
+          typename TObserver>
+inline void
+init(TraverserImpl<TJst, JstTraversalSpec<TSpec> > & me,
+     TObserver & observer)
+{
+    init(me, observer, SelectValidProxy());
+}
 
 // ----------------------------------------------------------------------------
 // Function goNext();
 // ----------------------------------------------------------------------------
 
-template <typename TJst, typename TSpec, typename TObserver,
-          typename TSize>
+template <typename TJst, typename TSpec,
+          typename TSize,
+          typename TObserver,
+          typename TProxySelector>
 inline void
-goNext(TraverserImpl<TJst, JstTraversalSpec<TSpec>, TObserver> & me,
-       TSize stepSize)
+advance(TraverserImpl<TJst, JstTraversalSpec<TSpec> > & me,
+        TSize stepSize,
+        TObserver & observer,
```

Comment:
Please make the sizes and positions also const. I know this is not done throughout seqan, but its a real pain, because in break overload-resolution if mixing const and non-const (so always use const if you can).